src/main/java/org/opentripplanner/graph_builder/module/osm/naming/SidewalkNamer.java

```diff
@@ -0,0 +1,304 @@
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.Envelope;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.LineString;
+import org.locationtech.jts.geom.MultiLineString;
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.operation.buffer.BufferParameters;
+import org.locationtech.jts.operation.distance.DistanceOp;
+import org.opentripplanner.framework.geometry.GeometryUtils;
+import org.opentripplanner.framework.geometry.HashGridSpatialIndex;
+import org.opentripplanner.framework.geometry.SphericalDistanceLibrary;
+import org.opentripplanner.framework.i18n.I18NString;
+import org.opentripplanner.framework.lang.DoubleUtils;
+import org.opentripplanner.framework.logging.ProgressTracker;
+import org.opentripplanner.graph_builder.module.osm.StreetEdgePair;
+import org.opentripplanner.graph_builder.services.osm.EdgeNamer;
+import org.opentripplanner.openstreetmap.model.OSMWithTags;
+import org.opentripplanner.street.model.edge.StreetEdge;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A namer that assigns names of nearby streets to sidewalks if they meet certain
+ * geometric similarity criteria.
+ * <p>
+ * The algorithm works as follows:
+ *  - for each sidewalk we look up (named) street edges nearby
+ *  - group those edges into groups where each edge has the same name
+ *  - draw a flat-capped buffer around the sidewalk, like this: https://tinyurl.com/4fpe882h
+ *  - check how much of a named edge group is inside the buffer
+ *  - remove those groups which are below MIN_PERCENT_IN_BUFFER
+ *  - take the group that has the highest percentage (as a proportion of the sidewalk length) inside
+ *    the buffer and apply its name to the sidewalk.
+ * <p>
+ * This works very well for OSM data where the sidewalk runs a parallel to the street and at each
+ * intersection the sidewalk is also split. It doesn't work well for sidewalks that go around
+ * the corner, like https://www.openstreetmap.org/way/1059101564. These cases are, however, detected
+ * by the above algorithm and the sidewalk name remains the same.
+ */
+public class SidewalkNamer implements EdgeNamer {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SidewalkNamer.class);
+  private static final int MAX_DISTANCE_TO_SIDEWALK = 50;
+  private static final double MIN_PERCENT_IN_BUFFER = .85;
+  private static final int BUFFER_METERS = 25;
+
+  private HashGridSpatialIndex<EdgeOnLevel> streetEdges = new HashGridSpatialIndex<>();
+  private Collection<EdgeOnLevel> unnamedSidewalks = new ArrayList<>();
+  private PreciseBuffer preciseBuffer;
+
+  @Override
+  public I18NString name(OSMWithTags way) {
+    return way.getAssumedName();
+  }
+
+  @Override
+  public void recordEdges(OSMWithTags way, StreetEdgePair pair) {
+    if (way.isSidewalk() && way.needsFallbackName() && !way.isExplicitlyUnnamed()) {
+      pair
+        .asIterable()
```

Comment:
No, the sidewalks don't need to have separate names for either side.

I just implemented the optimization that you've mentioned here and I'm sad to say that it did not have any effect on the processing time at all.

#### Before
```
(SidewalkNamer.java:121) Assigning names to sidewalks progress tracking complete. 109,380 done in 13s (8,296 per second).
```

#### After
```
(SidewalkNamer.java:119) Assigning names to sidewalks progress tracking complete. 54,690 done in 13s (4,158 per second).
``` 

If you feel strongly about this, I can include this but I feel we are getting into the area of diminishing returns here.

Reply:
On reflection the concept of fallback name which I introduced here probably does more harm than good. Whether something really needs a fallback name is quite subjective and cannot be derived from the source data.

For that reason I went with the follow method names:

- hasName
- hasNoName
- isExplicitlyUnnamed

This strictly states what can be seen from the source data and not some form of other judgement. I've also clarified the Javadoc.

You touched on another point which lead me into this in the first place: I had to google what "bogus name" meant, since it's not a commen word in British English. I would like to replace it with another term. Maybe we can come up with better name in a dev meeting.